test/RecurringPurchase.t.sol

```diff
@@ -0,0 +1,786 @@
+// SPDX-License-Identifier: BSD-3-Clause
+pragma solidity 0.8.23;
+
+import "forge-std/Test.sol";
+import "forge-std/console.sol";
+
+import {CodeJar} from "codejar/src/CodeJar.sol";
+
+import {QuarkStateManager} from "quark-core/src/QuarkStateManager.sol";
+import {QuarkWallet} from "quark-core/src/QuarkWallet.sol";
+
+import {QuarkMinimalProxy} from "quark-proxy/src/QuarkMinimalProxy.sol";
+
+import {RecurringPurchase} from "src/RecurringPurchase.sol";
+
+import {YulHelper} from "./lib/YulHelper.sol";
+import {SignatureHelper} from "./lib/SignatureHelper.sol";
+import {QuarkOperationHelper, ScriptType} from "./lib/QuarkOperationHelper.sol";
+
+import {IERC20} from "openzeppelin/token/ERC20/IERC20.sol";
+
+contract RecurringPurchaseTest is Test {
+    event SwapExecuted(
+        address indexed sender,
+        address indexed recipient,
+        address tokenIn,
+        address tokenOut,
+        uint256 amountIn,
+        uint256 amountOut,
+        bytes path
+    );
+
+    CodeJar public codeJar;
+    QuarkStateManager public stateManager;
+    QuarkWallet public walletImplementation;
+
+    uint256 alicePrivateKey = 0x8675309;
+    address aliceAccount = vm.addr(alicePrivateKey);
+    QuarkWallet aliceWallet; // see constructor()
+
+    bytes recurringPurchase = new YulHelper().getCode("RecurringPurchase.sol/RecurringPurchase.json");
+
+    // Contracts address on mainnet
+    address constant USDC = 0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48;
+    address constant WETH = 0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2;
+    address constant COMP = 0xc00e94Cb662C3520282E6f5717214004A7f26888;
+    // Uniswap router info on mainnet
+    address constant uniswapRouter = 0xE592427A0AEce92De3Edee1F18E0157C05861564;
+    // Price feeds
+    address constant ETH_USD_PRICE_FEED = 0x5f4eC3Df9cbd43714FE2740f5E3616155c5b8419; // Price is $1790.45
+    address constant USDC_USD_PRICE_FEED = 0x8fFfFfd4AfB6115b954Bd326cbe7B4BA576818f6;
+
+    constructor() {
+        // Fork setup
+        vm.createSelectFork(
+            string.concat(
+                "https://node-provider.compound.finance/ethereum-mainnet/", vm.envString("NODE_PROVIDER_BYPASS_KEY")
+            ),
+            18429607 // 2023-10-25 13:24:00 PST
+        );
```

Comment:
Note that we no longer control `node-provider.compound.finance` and eventually cannot rely on having access to bypass keys for tests.